invenio_rdm_records/config.py

```diff
@@ -261,6 +269,18 @@ def always_valid(identifier):
         "mostviewed",
         "mostdownloaded",
     ],
+    "query_parser_cls": QueryParser.factory(
+        mapping={
+            "internal_notes.note": RestrictedTerm(administration_permission),
```

Comment:
I would gladly align on using the correct permission here, I used admin for self explaining PR

Reply:
agreed with @ntarocco to settle on super-user permission for now